### ISSUE TITLE
add `data-testid` to locator builder

### ISSUE THREAD

**Issue by MagicDuck** [SeleniumHQ/selenium-ide#990]
## 🚀 Feature Proposal

Add `data-testid` as an element test id attribute to look for when recording.
The change is minimal:
```diff
@ packages/side-recorder/src/content/locator-builders.js:267 @ LocatorBuilders.prototype.preciseXPath = function(xpath, e) {
// e.g., 1st listed is top priority

LocatorBuilders.add('css:data-attr', function cssDataAttr(e) {
-  const dataAttributes = ['data-test', 'data-test-id']
+  const dataAttributes = ['data-test', 'data-test-id', 'data-testid']
```

## Motivation

`data-testid` is the convention in some existing codebases due to the fact that some tools work with it. For instance:
- React testing library: https://testing-library.com/docs/dom-testing-library/api-queries#bytestid
- cypress: https://docs.cypress.io/guides/references/best-practices.html 

Replacing all `data-testid` instances with `data-test` for example throughout the codebase and existing tests can be an obstacle to the adoption of selenium ide.

## Example

We are investigating using the new Selenium IDE to record our integration tests instead of using straight selenium here at [xMatters](https://www.xmatters.com/). So far, it looks pretty awesome apart from this small blocker :)

**Comment by corevo** [comment]
You are correct that replacing is near impossible, who knows what else will break.

In the meantime I can recommend adding that attribute additionally, and it would have the same data as the one you already have implemented.

**Comment by MagicDuck** [comment]
@corevo thanks for the response! So are you ok with me creating a PR with the change outlined above?

**Comment by Irrelon** [comment]
This definitely would be a good update. Would make using selenium IDE much better out of the box as it is then compatible with other tools in the JS ecosystem.

**Comment by Irrelon** [comment]
Either that, or allow the locators to be configurable per-project?

**Comment by toddtarsi** [comment]
@Irrelon - Configurability is good and all, but `data-testid` should definitely be the default top tier. Also, if I do this change, it will be v4 only sry

**Comment by tokyo-watcher** [comment]
Yes, please do this improvement.
I also created bug report for `if (attr) {`.
https://github.com/SeleniumHQ/selenium-ide/issues/1598

**Comment by tokyo-watcher** [comment]
@toddtarsi 
Thank you for your action on #1598.

Certainly, I confirmed that "data-test-id" can be used with 4.0.0-alpha.36 on Windows.
Also, I confirmed that "data-testid" can be used with 4.0.0-alpha.36 on Windows.

On the other hand, "data-testid" seems to have lower priority than "id" or "name".
Is there a plan to raise the priority of "data-testid"
to the same as "data-test" and "data-test-id" in this task #990?

**Comment by toddtarsi** [comment]
That’s a very reasonable tweakSent from my iPhoneOn Mar 15, 2023, at 9:43 PM, test20161010 ***@***.***> wrote:﻿
@toddtarsi
Thank you for your action on #1598.
Certainly, I confirmed that "data-test-id" can be used with 4.0.0-alpha.36 on Windows.
Also, I confirmed that "data-testid" can be used with 4.0.0-alpha.36 on Windows.
On the other hand, "data-testid" seems to have lower priority than "id" or "name".
Is there a plan to raise the priority of "data-testid"
to the same as "data-test" and "data-test-id" in this task #990?
